### PR TITLE
[ELY-916] Add ability to use specified Provider[] when creating KeyStore instance in KeyStoreCredentialStore

### PR DESCRIPTION
**Pull request by pskopek**
https://issues.jboss.org/browse/ELY-916

This will give us opportunity to use custom KeyStore (for example for FIPS certified CredentialStore).

### CODE REVIEW

**Comment by pskopek** [comment]
The "NoSuchAlgorithmException: 1.2.840.113549.1.7.1" in latest test runs are there due to JDK with no unlimited JCE policy.

**Comment by pskopek** [comment]
retest this please

**Comment by darranl** [comment]
I have submitted a PR to disable that test https://github.com/wildfly-security/wildfly-elytron/pull/656, do you think it is reasonable to find a way to enable it again or leave it off now?

File: src/main/java/org/wildfly/security/credential/store/impl/KeyStoreCredentialStore.java [hunk 1]
```diff
@@ -810,6 +815,23 @@ private void load(String type) throws CredentialStoreException {
         }
     }
 
+    private KeyStore getKeyStoreInstance(String type) throws CredentialStoreException {
+        if (providers != null) {
+            for (Provider p: providers) {
+                try {
+                    return KeyStore.getInstance(type, p);
+                } catch (KeyStoreException e) {
```
Comment (darranl) [inline src/main/java/org/wildfly/security/credential/store/impl/KeyStoreCredentialStore.java]: I am going to go ahead and include this PR but I think elsewhere we have some code that checks the capabiltiies of the Provider in advance to find the compatible one - may be worth looking for for here so we are not relying on error handling.

**Pull request merged**